pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -116,35 +113,29 @@ func (Plugin) OnOutboundCluster(env model.Environment, node model.Proxy, service
 	servicePort *model.Port, cluster *xdsapi.Cluster) {
 }
 
-func isRbacEnabled(store model.IstioConfigStore) (bool, error) {
+func isRbacEnabled(store model.IstioConfigStore) bool {
 	var configProto *rbacproto.RbacConfig
 	config := store.RbacConfig(RbacConfigName, kube.IstioNamespace)
 	if config != nil {
 		configProto = config.Spec.(*rbacproto.RbacConfig)
 	}
 	if configProto == nil {
-		return false, fmt.Errorf("couldn't find RbacConfig %s in namespace %s",
-			RbacConfigName, kube.IstioNamespace)
+		log.Debugf("rbac plugin disabled: No RbacConfig")
+		return false
 	}
 	// TODO(yangminzhu): Supports ON_WITH_INCLUSION and ON_WITH_EXCLUSION.
-	if configProto.Mode != rbacproto.RbacConfig_ON {
-		log.Debugf("rbac plugin disabled by rbacConfig: %v", *configProto)
-		return false, nil
+	if configProto.Mode == rbacproto.RbacConfig_ON {
+		return true
 	}
 
-	return true, nil
+	log.Debugf("rbac plugin disabled by RbacConfig: %v", *configProto)
+	return false
 }
 
 // buildHTTPFilter builds the RBAC http filter that enforces the access control to the specified
 // service which is co-located with the sidecar proxy.
 func buildHTTPFilter(hostName model.Hostname, store model.IstioConfigStore) (*http_conn.HttpFilter, error) {
-	service := string(hostName)
-	split := strings.Split(service, ".")
-	if len(split) < 2 {
-		return nil, fmt.Errorf("failed to extract namespace from service: %s", service)
-	}
-	namespace := split[1]
-
+	namespace := hostName.Namespace()
 	roles := store.ServiceRoles(namespace)
 	if roles == nil {
 		return nil, fmt.Errorf("failed to get ServiceRoles in namespace %s", namespace)
```

Comment:
I'm not sure if error is needed, or the message accurate. If customer doesn't define any service role, then of course we have no roles, but it's not a `failure` get. May be change message to "No service roles found in namespace ..." instead.

Reply:
Refactored to remove the error.
Originally the store.ServiceRole should only return nil when an error happened but return an empty slice if no service roles found.